### PR TITLE
[Refactor] Remove the Venmo payment option

### PR DESCRIPTION
**Pull request by mountiny**
### Details

As part of the Refactoring, we have decided to remove the Venmo payment option from the NewDot.

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/216466

### Tests

- [x] Verify that no errors appear in the JS console

To verify secondary logins locally, you should use the `/script/clitools.sh validator:secondarylogin` script. It might be best to add the secondary logins from OldDot

#### Test Set 1 - Venmo payment option removed

1. Have 2 acocunts, Account A and Account B both with added and verified phone secondary login and paypal account added
2. Attempt to Send money from Account A to account B
3. Make sure the payment options do not include Venmo
image

#### Test Set 2 - no Venmo hints on the settings page

1. Have an account which does not have a phone number set up
2. Go to Settings > Profile
3. Make sure that there is no Note written under the Add phone button number

#### Test Set 3 - no Venmo hints on the settings page when phone number is already set up

1. Have an account which has phone number set up and verified
2. Go to Settings > Profile
3. Make sure that there is no Note written under the phone number:
image

#### Test Set 4 - confirm payment flows work as expected

Confirm all the payment flows (sending/requesting money) Elsewhere, Paypal and Wallet. 

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [ ] Android / native
    - [x] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

- [ ] Verify that no errors appear in the JS console

### Screenshots

The settings page - Phone is set up and no note is shown:
image

#### Web

image

#### Mobile Web

image

#### Desktop

image

#### iOS

image

image

### CODE REVIEW

**Comment by mountiny** [comment]
Asked in [Slack](https://expensify.slack.com/archives/C02HWMSMZEC/p1656677281794089) if we would like to remove the Notes in Settings page when user does not have phone number setup since they mention Venmo. We will remove them and new notes can be added in future as a positive motivation to add phone number.

I have removed all the code related to Venmo as far as I know. I havent been able to test yet though and get screenshots, gonan have to do that later so.

**Comment by mountiny** [comment]
Just having some problems  with set up when trying to test locally so I can get screenshots. But hopefully should be resolved tomorrow

**Comment by mountiny** [comment]
@Luke9389 Tested and it should be all fine. I have just been having some problems with the android emulator, I will add screens tomorrow but there is no reason it should not work there.

**Comment by mountiny** [comment]
Adding you Jules for a review as we work on this mini design doc together.

File: src/components/Icon/Expensicons.js [hunk 1]
```diff
@@ -70,7 +70,6 @@ import Trashcan from '../../../assets/images/trashcan.svg';
 import UpArrow from '../../../assets/images/arrow-up.svg';
 import Upload from '../../../assets/images/upload.svg';
 import Users from '../../../assets/images/users.svg';
-import Venmo from '../../../assets/images/venmo.svg';
```
[the remark this answers is not in the document]
Comment (mountiny) [reply src/components/Icon/Expensicons.js:73]: Good call, it has not been. Updated.

**Comment by mountiny** [comment]
I am having problems running the Android locally. I dont think that should be a blocker though

**Comment by Justicea83** [comment]
I learnt the android build is broken, https://github.com/Expensify/App/pull/9649#issuecomment-1176623162

**Comment by mountiny** [comment]
Might be related to it, but I don't think this is it for me, I have intel and I haven't run the Android emulator in a while and it is giving me some weird errors, but havent prioritized fixing it.

File: src/components/SettlementButton.js [hunk 2]
```diff
@@ -64,63 +57,11 @@ class SettlementButton extends React.Component {
...
-
-    /**
-     * Adds Venmo, if available, as the second option in the menu of payment options
-     */
-    addVenmoPaymentOptionToMenu() {
-        if (this.props.currency !== CONST.CURRENCY.USD || !this.doesRecipientHaveValidPhoneLogin()) {
-            return;
-        }
-
-        this.checkVenmoAvailabilityPromise = makeCancellablePromise(isAppInstalled('venmo'));
```
Comment (Julesssss) [inline src/components/SettlementButton.js:103]: Is `isAppInstalled` used anywhere else? I think we only added it for Venmo.

Comment (mountiny) [reply src/components/SettlementButton.js:103]: Gonna have a look! Thanks!

Comment (mountiny) [reply src/components/SettlementButton.js:103]: Removed!

**Comment by mountiny** [comment]
@Luke9389 @Julesssss @Justicea83 Updated, thanks for having another look!

**Review by Luke9389 — COMMENTED** [review]
Quick Q

File: src/components/IOUConfirmationList.js [hunk 3]
```diff
@@ -370,7 +370,6 @@ class IOUConfirmationList extends Component {
                             isLoading={isLoading}
                             onPress={this.confirm}
                             shouldShowPaypal={Boolean(recipient.payPalMeAddress)}
-                            recipientPhoneNumber={recipient.phoneNumber}
```
[the remark this answers is not in the document]
Comment (mountiny) [reply src/components/IOUConfirmationList.js:373]: @Luke9389 Correct, the phone number in this flow has only been used for the Venmo feature.

**Review by Luke9389 — APPROVED** [review]

File: src/components/IOUConfirmationList.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Luke9389) [reply src/components/IOUConfirmationList.js:373]: 👍 Thanks for the explanation!

**Comment by mountiny** [comment]
bumping @Julesssss @Justicea83

**Review by Julesssss — APPROVED** [review]
I only reviewed the code and tested web, as I was just CC'd and not a reviewer. But it looks good to me 👍 

It took me a while to find the secondary login validator. Maybe you could add it to the instructions: `/script/clitools.sh validator:secondarylogin`

Screenshot 2022-07-12 at 12 55 03

**Comment by isagoico** [comment]
@mountiny Hey! We have some [regression TCs that involve Venmo payments.](https://expensify.testrail.io/index.php?/suites/view/518&group_by=cases:section_id&group_order=asc&display_deleted_cases=0&group_id=140068) These will be blocked with this change. Just wanted to double confirm with you that those can be removed from the regression run before taking action.

**Comment by mountiny** [comment]
Correct, any tests related to Venmo flow in NewDot can be removed. Thank you very much! @isagoico

**Pull request merged**